pkg/agent/plugin/workloadattestor/docker/docker_test.go

```diff
@@ -31,57 +31,43 @@ const (
 
 func TestDockerSelectors(t *testing.T) {
 	tests := []struct {
-		desc                string
-		mockContainerLabels map[string]string
-		mockEnv             []string
-		mockImageID         string
-		requireResult       func(*testing.T, *workloadattestorv0.AttestResponse)
+		desc                 string
+		mockContainerLabels  map[string]string
+		mockEnv              []string
+		mockImageID          string
+		expectSelectorValues []string
 	}{
 		{
 			desc:                "single label; single env",
 			mockContainerLabels: map[string]string{"this": "that"},
 			mockEnv:             []string{"VAR=val"},
-			requireResult: func(t *testing.T, res *workloadattestorv0.AttestResponse) {
-				require.Len(t, res.Selectors, 2)
-				require.Equal(t, "docker", res.Selectors[0].Type)
-				require.Equal(t, "label:this:that", res.Selectors[0].Value)
-				require.Equal(t, "docker", res.Selectors[1].Type)
-				require.Equal(t, "env:VAR=val", res.Selectors[1].Value)
+			expectSelectorValues: []string{
+				"env:VAR=val",
+				"label:this:that",
 			},
 		},
 		{
 			desc:                "many labels; many env",
 			mockContainerLabels: map[string]string{"this": "that", "here": "there", "up": "down"},
 			mockEnv:             []string{"VAR=val", "VAR2=val"},
-			requireResult: func(t *testing.T, res *workloadattestorv0.AttestResponse) {
-				require.Len(t, res.Selectors, 5)
-				expectedSelectors := map[string]struct{}{
-					"label:this:that":  {},
-					"label:here:there": {},
-					"label:up:down":    {},
-					"env:VAR=val":      {},
-					"env:VAR2=val":     {},
-				}
-				for _, selector := range res.Selectors {
-					require.Equal(t, "docker", selector.Type)
-					require.Contains(t, expectedSelectors, selector.Value)
-				}
+			expectSelectorValues: []string{
+				"env:VAR2=val",
+				"env:VAR=val",
+				"label:here:there",
+				"label:this:that",
+				"label:up:down",
 			},
 		},
 		{
-			desc:                "no labels or env for container",
-			mockContainerLabels: map[string]string{},
-			requireResult: func(t *testing.T, res *workloadattestorv0.AttestResponse) {
-				require.Len(t, res.Selectors, 0)
-			},
+			desc:                 "no labels or env for container",
+			mockContainerLabels:  map[string]string{},
+			expectSelectorValues: nil,
```

Comment:
not sure if we may expect nil? how about we returns an empty array instead?

Reply:
You can't return an empty array over gRPC, so I feel `nil` is more precise? If we allow an empty array here, we'd have to do special handling in the assertSelectorsEqual method...

Reply:
makes sense